inspire_crawler/tasks.py

```diff
@@ -108,16 +133,27 @@ def _extract_results_data(results_path):
     if results_data is None:
         results_data = _extract_results_data(results_path)
 
-    for record in results_data:
+    for result in results_data:
+        crawl_result = _check_crawl_result_format(result)
+        record = crawl_result['record']
+        crawl_errors = crawl_result['errors']
+
         current_app.logger.debug(
             'Parsing record: {}'.format(record)
         )
-        record_error = record.get('error')
-        if record_error is not None:
-            obj = workflow_object_class.create(data=record['xml_record'])
+        if crawl_errors:
+            obj = workflow_object_class.create(data=record)
             obj.status = ObjectStatus.ERROR
-            obj.extra_data['_error_msg'] = record_error
-            obj.extra_data['callback_result'] = record['traceback']
+
+            crawled_data = crawl_result['source_data']
+            obj.extra_data['crawled_data'] = crawled_data
+
+            obj.extra_data['_error_msg'] = [
+                err['exception'] for err in crawl_errors
+            ]
```

Comment:
This should be a string no?

Reply:
We have a list of errors, should I take just the first one?

Reply:
this is the string that is shown in the workflow footer when there's an error, I'd use something like `\n`.join() thingie.